test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,103 @@
+	. "github.com/onsi/gomega"
+	clusterv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing live iso [live-iso]", func() {
+	liveIsoTest()
+})
+
+func liveIsoTest() {
+	BeforeEach(func() {
+		validateGlobals(specName)
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+
+	It("Should create a target cluster node with live iso", func() {
+		controlPlaneMachineCount = 1
+		workerMachineCount = 1
+
+		workloadClusterTemplate := clusterctl.ConfigCluster(ctx, clusterctl.ConfigClusterInput{
+			KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+			ClusterctlConfigPath:     clusterctlConfigPath,
+			Flavor:                   "live-iso",
+			Namespace:                namespace,
+			ClusterName:              clusterName,
+			KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+			ControlPlaneMachineCount: &controlPlaneMachineCount,
+			WorkerMachineCount:       &workerMachineCount,
+			InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+			LogFolder:                clusterctlLogFolder,
+		})
+
+		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+		Expect(bootstrapClusterProxy.Apply(ctx, workloadClusterTemplate)).To(Succeed())
+
+		By("Waiting for the control plane to be running and worker node to be in provisioning state")
+		Eventually(func() (int, error) {
+			n := 0
+			machineList := &clusterv1alpha4.MachineList{}
+			if err := bootstrapClusterProxy.GetClient().List(ctx, machineList, client.InNamespace(namespace), client.MatchingLabels{clusterv1.ClusterLabelName: clusterName}); err == nil {
+				for _, machine := range machineList.Items {
+					if isWorkerNode(machine) && strings.EqualFold(machine.Status.Phase, "provisioning") {
+						n++
+					} else if strings.EqualFold(machine.Status.Phase, "running") {
+						n++
+					}
+				}
+			}
+			return n, nil
+		}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Equal(2))
+
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+		controlplaneM3Machines, workerM3Machines := getMetal3Machines(ctx, bootstrapClient, clusterName, namespace)
+		Expect(controlplaneM3Machines).To(HaveLen(int(controlPlaneMachineCount)))
+
+		getBmhFromM3Machine := func(m3Machine infrav1.Metal3Machine) (result bmov1alpha1.BareMetalHost) {
+			Expect(bootstrapClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: metal3MachineToBmhName(m3Machine)}, &result)).To(Succeed())
+			return result
+		}
+
+		workerBmh := getBmhFromM3Machine(workerM3Machines[0])
+		vmName := bmhToVMName(workerBmh)
+		serialLogFile := fmt.Sprintf("/var/log/libvirt/qemu/%s-serial0.log", vmName)
+		time.Sleep(5 * time.Minute)
```

Comment:
Why sleeping here ? can replace that directly with a loop check is the logs reached what we are expecting ?